### PR TITLE
WE-321 Configure page routing for support docs

### PR DESCRIPTION
**Pull request by jonambas**
**What Changed**
- Adds `index.tsx` and `[...slug].tsx` routes for `docs`
- Adds a redirect from `/` to `/docs`
- Abstracts out layout into a `Layout` component that is shared between docs and momentum
- Abstracts out `ReactMarkdown` component so that is shared between docs and momentum
- Moves all content from `/content/support` to `/content/docs` to reflect existing routing

**How to Verify**
- Run the site locally or visit the deploy preview
- Visit `/` to verify the redirect
- Visit `/docs` to verify the home page route
- Visit `/docs/getting-started/getting-started-sparkpost` to verify single post page content is loading correctly
- Additionally, visit `/momentum` pages to verify the abstractions worked correctly

### CODE REVIEW

File: lib/api.ts [hunk 1]
```diff
@@ -44,18 +44,18 @@ export const getAllCategoryPostPaths = (category: CategoryOption): string[] => {
...
 
   // If file exists, it is not an index page.
   if (fs.existsSync(filePath)) {
-    return matter(fs.readFileSync(filePath, 'utf8'));
+    return { isIndex: false, ...matter(fs.readFileSync(filePath, 'utf8')) };
   }
 
   if (fs.existsSync(indexPath)) {
-    return matter(fs.readFileSync(indexPath, 'utf8'));
+    return { isIndex: true, ...matter(fs.readFileSync(indexPath, 'utf8')) };
```
Comment (jonambas) [inline lib/api.ts:58]: Added this boolean so we can use it to differentiate single post vs. index page layouts

File: components/site/docsLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Layout from './layout';
+import DocsNavigation from './docsNavigation';
+
+type LayoutProps = {
+  children?: React.ReactNode;
+};
+
+const DocsLayout = (props: LayoutProps): JSX.Element => {
+  const { children } = props;
+  return <Layout navigationComponent={<DocsNavigation />}>{children}</Layout>;
```
Comment (jonambas) [inline components/site/docsLayout.tsx:10]: Technically, don't really need this component at all and could just add this code to the `[...slug]` page

**Review by EvanYoung-SP — APPROVED** [review]
LGTM! Had one clarifying question, but it isn't a change request. Great work, man!

File: components/site/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            href="#main-content"
+            position="absolute"
+            display="inline-block"
+            p="200"
+            top="0"
+            left="0"
+            bg="blue.700"
+          >
+            Skip to main content
+          </StyledSkipToContent>
```
[the remark this answers is not in the document]
Comment (jonambas) [reply components/site/layout.tsx:39]: If you focus the browser's address bar then hit tab it should show up. It's intentionally hidden and only meant for screen readers

**Pull request merged**